connectivity/check/frr.go

```diff
@@ -250,41 +254,56 @@ func RenderFRRBGPPeeringConfig(t *Test, params FRRBGPPeeringParams) string {
 }
 
 // WaitForFRRBGPNeighborsState waits until provided list of BGP peers reach the provided state
-// on the provided FRR pod.
-func WaitForFRRBGPNeighborsState(ctx context.Context, t *Test, frrPod *Pod, expPeers []netip.Addr, expState string) {
+// on the provided FRR pod and returns detailed state information of all peers.
+func WaitForFRRBGPNeighborsState(ctx context.Context, t *Test, frrPod *Pod, expPeers []netip.Addr, expState string) map[string]FRRBGPNeighborInfo {
 	w := wait.NewObserver(ctx, wait.Parameters{Timeout: 30 * time.Second})
 	defer w.Cancel()
 
-	ensureBGPNeighborsState := func() error {
+	ensureBGPNeighborsState := func() (map[string]FRRBGPNeighborInfo, error) {
 		stdout := RunFRRCommand(ctx, t, frrPod, "show bgp neighbor json")
 		entries := map[string]FRRBGPNeighborInfo{}
 		err := json.Unmarshal(stdout, &entries)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		if len(entries) < len(expPeers) {
-			return fmt.Errorf("expected %d peers, got %d", len(expPeers), len(entries))
+			return nil, fmt.Errorf("expected %d peers, got %d", len(expPeers), len(entries))
 		}
 		for _, peer := range expPeers {
 			frrPeer, exists := entries[peer.String()]
 			if !exists {
-				return fmt.Errorf("missing peer %s", peer.String())
+				return nil, fmt.Errorf("missing peer %s", peer.String())
 			}
 			if frrPeer.BGPState != expState {
-				return fmt.Errorf("peer %s: expected %s state, got %s", peer, expState, entries[peer.String()].BGPState)
+				return nil, fmt.Errorf("peer %s: expected %s state, got %s", peer, expState, entries[peer.String()].BGPState)
 			}
 		}
-		return nil
+		return entries, nil
 	}
 
 	for {
-		if err := ensureBGPNeighborsState(); err != nil {
+		entries, err := ensureBGPNeighborsState()
+		if err != nil {
 			if err := w.Retry(err); err != nil {
 				t.Fatalf("Failed to ensure FRR BGP neighbor states: %v", err)
 			}
 			continue
 		}
-		return
+		return entries
+	}
+}
+
+// AssertFRRBGPNeighborTimers asserts that peering connections of the provided neighbors filtered by checkNeighbors
+// use the provided BGP timer intervals.
+func AssertFRRBGPNeighborTimers(t *Test, neighbors map[string]FRRBGPNeighborInfo, checkNeighbors []netip.Addr, keepAliveSeconds, holdTimeSeconds int32) {
+	for _, addr := range checkNeighbors {
+		neighbor := neighbors[addr.String()]
```

Comment:
We should check for existence in the neighbors map. 

This is probably not possible in the way the test is using AssertFRRBGPNeighborTimers.